### PR TITLE
Issue 2541: hasMonth() and hasMonthValue()

### PR DESCRIPTION
**Pull request by nith2001**
#### Check List:
* Fixes partially #2541
* Unit tests : YES
* Javadoc with a code example (on API only) : YES
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by nith2001** [comment]
That's a lot of failures....Any reason why there would be this many failures across versions? New to large codebase Java programming, so could use some advice.

**Comment by scordio** [comment]
Some files are lacking the license header:

```
[INFO] --- license-maven-plugin:4.1:check (default) @ assertj-core ---
[INFO] Checking licenses...
Warning:  Missing header in: /home/runner/work/assertj-core/assertj-core/src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java
Warning:  Missing header in: /home/runner/work/assertj-core/assertj-core/src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java
[INFO] ------------------------------------------------------------------------
```

You can run:

`./mvnw license:format`

to get them fixed automatically.

**Comment by nith2001** [comment]
Ok, the header fixed most of the tests, its just one failure which I don't quite know how to interpret.

**Comment by scordio** [comment]
That's a known issue of pitest, you can ignore it.

**Comment by scordio** [comment]
That's fine: forks don't have access to the Sonar token so that job gets skipped.

File: src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java [hunk 1]
```diff
@@ -480,6 +481,56 @@ public SELF hasYear(int year) {
...
+   * assertThat(LocalDate.of(2000, 12, 31)).hasMonth(11);</code></pre>
+   *
+   * @param month the given month's value between 1 and 12 inclusive.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code LocalDate} is {@code null}.
+   * @throws AssertionError if the actual {@code LocalDate} is not in the given month.
+   *
+   * @since 3.23.0
+   */
+  public SELF hasMonth(int month) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java:526]: this should be `hasMonthValue`

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import java.time.LocalDate;
+import java.time.Month;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldHaveDateField.shouldHaveDateField;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+public class LocalDateAssert_hasMonthEnum_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java:26]: no need for the class to be public
rename to `LocalDateAssert_hasMonth_Test`

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  @Test
+  void should_pass_if_actual_is_in_given_month() {
+    // GIVEN
+    LocalDate actual = LocalDate.of(2021, 2, 22);
+    // WHEN/THEN
+    then(actual).hasMonth(Month.FEBRUARY);
+  }
+
+  @Test
+  void should_pass_if_actual_is_in_given_month2() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java:37]: this test is not needed, it's already covered by `should_pass_if_actual_is_in_given_month`

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    LocalDate actual = LocalDate.of(2022, 1, 1);
+    Month wrongMonth = Month.DECEMBER;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).hasMonth(wrongMonth));
+    // THEN
+    then(assertionError).hasMessage(shouldHaveDateField(actual, "month", wrongMonth.getValue()).create());
+  }
+
+  @Test
+  void should_fail_if_actual_is_not_in_given_month2() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthEnum_Test.java:57]: remove this test as it is covered by `should_fail_if_actual_is_not_in_given_month`

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import java.time.LocalDate;
+import java.time.Month;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldHaveDateField.shouldHaveDateField;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+public class LocalDateAssert_hasMonthInt_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java:26]: rename to `LocalDateAssert_hasMonthValue_Test`
remove `public`

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  @Test
+  void should_pass_if_actual_is_in_given_month() {
+    // GIVEN
+    LocalDate actual = LocalDate.of(2021, 2, 22);
+    // WHEN/THEN
+    then(actual).hasMonth(2);
+  }
+
+  @Test
+  void should_pass_if_actual_is_in_given_month2() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java:38]: remove

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    LocalDate actual = LocalDate.of(2022, 1, 1);
+    int wrongMonth = 12;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).hasMonth(wrongMonth));
+    // THEN
+    then(assertionError).hasMessage(shouldHaveDateField(actual, "month", wrongMonth).create());
+  }
+
+  @Test
+  void should_fail_if_actual_is_not_in_given_month2() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasMonthInt_Test.java:58]: remove

File: src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasYear_Test.java [hunk 8]
```diff
@@ -53,4 +53,5 @@ void should_fail_if_actual_is_null() {
     then(assertionError).hasMessage(actualIsNull());
   }
 
+
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/localdate/LocalDateAssert_hasYear_Test.java:56]: revert this change

**Review by joel-costigliola — COMMENTED** [review]
Thanks @nith2001, first round of code review done.

**Comment by nith2001** [comment]
Resolved all notes from @joel-costigliola

**Comment by nith2001** [comment]
Ok, tests are all passing, fixed all of the things from the review @joel-costigliola

**Comment by joel-costigliola** [comment]
Integrated thanks @nith2001!

could you avoid merging `main` in your branch and rebase instead for the next PR, it makes it easier to integrate as we prefer squashing PR into one commit

**Comment by nith2001** [comment]
Hi @joel-costigliola ! I've never really rebased before, but I just did some Google searching and definitely seems simpler. How would I do that in this case? is there some rebase option I can select in the PR?

**Comment by joel-costigliola** [comment]
That's an opportunity to learn something new then ;-)
So what you would do is rebase the branch corresponding to your PR on `main`, the end result would be to "move" your branch commits like you had started from the latest main commit.

But don't worry too much about it, if you struggle too much do a merge.

**Comment by nith2001** [comment]
Hmmm, I see. I'll try to do that on that on the next issue I took.
Would it be doing something like this, and then submitting a pull request?

Screen Shot 2022-04-16 at 1 21 53 AM

File: src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java [hunk 9]
```diff
@@ -480,6 +481,56 @@ public SELF hasYear(int year) {
...
+    if (!actual.getMonth().equals(month)) {
+      throw Failures.instance().failure(info, shouldHaveDateField(actual, "month", month.getValue()));
+    }
+    return myself;
+  }
+
+  /**
+   * Verifies that actual {@code LocalDate} is in the given month.
+   * <p>
+   * Example:
```
Comment (Zhou-Yicheng) [inline src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java:512]: should be hasMonthValue() here

File: src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java [hunk 10]
```diff
@@ -480,6 +481,56 @@ public SELF hasYear(int year) {
...
+   * assertThat(LocalDate.of(2000, 12, 31)).hasMonth(11);</code></pre>
+   *
+   * @param month the given month's value between 1 and 12 inclusive.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code LocalDate} is {@code null}.
+   * @throws AssertionError if the actual {@code LocalDate} is not in the given month.
+   *
+   * @since 3.23.0
+   */
+  public SELF hasMonthValue(int month) {
```
Comment (Zhou-Yicheng) [inline src/main/java/org/assertj/core/api/AbstractLocalDateAssert.java:526]: should be hasMonthValue() in the example
